datalake/src/test/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreActionsTest.java

```diff
@@ -0,0 +1,144 @@
+
+    @Mock
+    private FlowLogService flowLogService;
+
+    @Mock
+    private FlowChainLogService flowChainLogService;
+
+    @Test
+    public void testRestoreWithNoFlowChain() throws Exception {
+        when(sdxBackupRestoreService.triggerDatalakeRestore(eq(OLD_SDX_ID), any(), any(), eq(USER_CRN)))
+                .thenReturn(new DatalakeDrStatusResponse(BACKUP_ID, State.STARTED, Optional.empty()));
+        when(flowLogService.getLastFlowLog(anyString())).thenReturn(Optional.of(new FlowLog()));
+        when(flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(any())).thenReturn(Optional.empty());
+
+        DatalakeTriggerRestoreEvent event = new DatalakeTriggerRestoreEvent(DATALAKE_TRIGGER_RESTORE_EVENT.event(), OLD_SDX_ID, DATALAKE_NAME,
+                USER_CRN, null, BACKUP_LOCATION, null, DatalakeRestoreFailureReason.RESTORE_ON_RESIZE);
+        AbstractAction action = (AbstractAction) underTest.triggerDatalakeRestore();
+        initActionPrivateFields(action);
+        AbstractActionTestSupport testSupport = new AbstractActionTestSupport(action);
+        SdxContext context = SdxContext.from(new FlowParameters(FLOW_ID, FLOW_ID, null), event);
+        testSupport.doExecute(context, event, new HashMap());
+
+        verify(reactorEventFactory, times(1)).createEvent(any(), any(DatalakeDatabaseRestoreStartEvent.class));
+    }
+
+    @Test
+    public void testGetNewSdxIdForResize() throws Exception {
+
+        SdxCluster sdxCluster = genCluster();
+        when(sdxService.getByNameInAccount(eq(USER_CRN), eq(DATALAKE_NAME))).thenReturn(sdxCluster);
+        when(sdxBackupRestoreService.triggerDatalakeRestore(eq(NEW_SDX_ID), any(), any(), eq(USER_CRN)))
+                .thenReturn(new DatalakeDrStatusResponse(BACKUP_ID, State.STARTED, Optional.empty()));
+        when(flowLogService.getLastFlowLog(anyString())).thenReturn(Optional.of(new FlowLog()));
+        FlowChainLog flowChainLog = new FlowChainLog();
+        flowChainLog.setFlowChainType(DatalakeResizeFlowEventChainFactory.class.getName());
+        when(flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(any())).thenReturn(Optional.of(flowChainLog));
+
+        DatalakeTriggerRestoreEvent event = new DatalakeTriggerRestoreEvent(DATALAKE_TRIGGER_RESTORE_EVENT.event(), OLD_SDX_ID, DATALAKE_NAME,
+                USER_CRN, null, BACKUP_LOCATION, null, DatalakeRestoreFailureReason.RESTORE_ON_RESIZE);
+        AbstractAction action = (AbstractAction) underTest.triggerDatalakeRestore();
+        initActionPrivateFields(action);
+        AbstractActionTestSupport testSupport = new AbstractActionTestSupport(action);
+        SdxContext context = SdxContext.from(new FlowParameters(FLOW_ID, FLOW_ID, null), event);
+        testSupport.doExecute(context, event, new HashMap());
+
+        verify(reactorEventFactory, times(1)).createEvent(any(), any(DatalakeDatabaseRestoreStartEvent.class));
+
+    }
+
```

Comment:
What are the tests verifying exactly?

Reply:
in particular we have an expectation of which (NEW_SDX_ID,OLD_SDX_ID) should be used when calling the resize. since that's the crux of the bug which was found

Reply:
I was looking for the same.
Can you point me the code the line where this validation is done?

Reply:
it's specified in Line 88 and 109, if they aren't passed in as expected the mocked services then throw an exception saying such